### PR TITLE
Add docs for providing release version matrix

### PR DESCRIPTION
**Pull request by nicolehanjing**
**What this PR does / why we need it**:
Add docs for providing release version matrix among Kubernetes Version, vSphere Cloud Provider Release Version and Cloud Provider Branch

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes https://github.com/kubernetes/cloud-provider-vsphere/issues/457

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,6 +8,22 @@ This project replaces the deprecated in-tree vSphere cloud provider located with
...
+
+Version matrix:
+
+| Kubernetes Version | vSphere Cloud Provider Release Version | Cloud Provider Branch |
+| ----------- | ----------- | ----------- |
+| v1.20.6 | v1.20.0-alpha.1 | master |
+| v1.19.7 | v1.19.0-beta.0 | release-1.19 |
+| v1.18.12 | v1.18.0-beta.0 | release-1.18 |
+| v1.18.12 | v1.18.0-alpha.1 | release-1.18 |
+| v1.18.6 | v1.18.0-alpha.0 | release-1.18 |
```
[the remark this answers is not in the document]
Comment (nicolehanjing) [reply README.md:25]: agree, updated!

**Review by lubronzhan — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -8,6 +8,20 @@ This project replaces the deprecated in-tree vSphere cloud provider located with
...
+
+In the future, the major and minor versions of releases should be equivalent to the compatible upstream Kubernetes release, and the patch version is used for bug fixes pertaining to specific Kubernetes releases. See [the external cloud provider versioning KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/1771-versioning-policy-for-external-cloud-providers) for more details.
+
+Version matrix:
+
+| Kubernetes Version | vSphere Cloud Provider Release Version | Cloud Provider Branch |
+| ----------- | ----------- | ----------- |
+| v1.20.X | v1.20.0 | release-1.20 |
+| v1.19.X | v1.19.0 | release-1.19 |
+| v1.18.X | v1.18.0 | release-1.18 |
```
Comment (andrewsykim) [inline README.md:23]: `v1.18.1` for `vSpere Cloud Provider Release Version`  or v1.18.X (same with the other ones)

**Review by andrewsykim — COMMENTED** [review]
just a minor nit

File: README.md [hunk 3]
```diff
@@ -8,6 +8,20 @@ This project replaces the deprecated in-tree vSphere cloud provider located with
...
+In the future, the major and minor versions of releases should be equivalent to the compatible upstream Kubernetes release, and the patch version is used for bug fixes pertaining to specific Kubernetes releases. See [the external cloud provider versioning KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-cloud-provider/1771-versioning-policy-for-external-cloud-providers) for more details.
+
+Version matrix:
+
+| Kubernetes Version | vSphere Cloud Provider Release Version | Cloud Provider Branch |
+| ----------- | ----------- | ----------- |
+| v1.20.X | v1.20.X | release-1.20 |
+| v1.19.X | v1.19.X | release-1.19 |
+| v1.18.X | v1.18.X | release-1.18 |
+
```
Comment (nicolehanjing) [inline README.md:24]: @andrewsykim updated, PTAL :)

**Review by andrewsykim — COMMENTED** [review]
/approve
/lgtm

Thanks!

**Pull request merged**